Common/Model.cs

```diff
@@ -0,0 +1,143 @@
+using TextureWrapMode = OpenTK.Graphics.OpenGL4.TextureWrapMode;
+using AssimpMesh = Assimp.Mesh;
+
+namespace LearnOpenTK.Common
+{
+    public class Model
+    {
+        // model data
+        private List<Mesh> meshes;
+
+        // constructor, expects a filepath to a 3D model.
+        public Model(string path)
+        {
+            LoadModel(path);
+        }
+
+        // draws the model, and thus all its meshes
+        public void Draw()
+        {
+            for (int i = 0; i < meshes.Count; i++)
+                meshes[i].Draw();
+        }
+
+        // loads a model with supported ASSIMP extensions from file and stores the resulting meshes in the meshes list.
+
+        private void LoadModel(string path)
+        {
+            //Create a new importer
+            AssimpContext importer = new AssimpContext();
+
+            //This is how we add a logging callback 
+            LogStream logstream = new LogStream(delegate (String msg, String userData)
+            {
+                Console.WriteLine(msg);
+            });
+            logstream.Attach();
+
+            //Import the model. All configs are set. The model
+            //is imported, loaded into managed memory. Then the unmanaged memory is released, and everything is reset.
+            Scene scene = importer.ImportFile(path, PostProcessSteps.Triangulate);
+
+            // check for errors
+            if (scene == null || scene.SceneFlags.HasFlag(SceneFlags.Incomplete) || scene.RootNode == null)
+            {
+                Console.WriteLine("Unable to load model from: " + path);
+                return;
+            }
+            
+            //Reset the meshes and textures
+            meshes = new List<Mesh>();
+
+            //Set the scale of the model
+            float scale = 1/200.0f;
+            Matrix4x4 scalingMatrix = new Matrix4x4(scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, 1);
+
+            // process ASSIMP's root node recursively. We pass in the scaling matrix as the first transform
+            ProcessNode(scene.RootNode, scene, scalingMatrix);
+
+            importer.Dispose();
+        }
+
+        // processes a node in a recursive fashion. Processes each individual mesh located at the node and repeats this process on its children nodes (if any).
+        private void ProcessNode(Node node, Scene scene, Matrix4x4 parentTransform)
+        {
+            //Multiply the transform of each node by the node of the parent, this will place the meshes in the correct relative location
+            Matrix4x4 transform = node.Transform*parentTransform;
+
+            // process each mesh located at the current node
+            for (int i = 0; i < node.MeshCount; i++)
+            {
+                // the node object only contains indices to index the actual objects in the scene. 
+                // the scene contains all the data, node is just to keep stuff organized (like relations between nodes).
```

Comment:
This description reads a bit weird, but we should say something about the scene hierarchy.

Reply:
I gave an attempt at rewording this. The original wording was from the source code give for the Model class on [learnopengl.com](https://learnopengl.com/code_viewer_gh.php?code=includes/learnopengl/model.h).